R/latent_individual.R

```diff
@@ -138,94 +195,50 @@ epidist_formula.epidist_latent_individual <- function(data, delay_central = ~ 1,
+  formula[[mu_index]] <- stats::update(
+    formula[[mu_index]],
+    delay_central | vreal(obs_t, pwindow_upr, swindow_upr) ~ .
   )
+  formula <- do.call(brms::bf, formula)
+  return(formula)
 }
 
 #' Define priors for the model
 #'
-#' We provide suggested priors for the intercepts of the `meanlog` and `sdlog`
-#' linear predictors. These priors are weakly informative in that they prevent
-#' very large delays on the real scale. In particular:
-#'
-#' * `eta_meanlog ~ normal(2, 0.5)`
-#' * `eta_sdlog ~ normal(0, 0.5)`
-#'
-#' Note that the link function used for `meanlog` is a constant, and the link
-#' function used for `sdlog` is a logarithm, such that the expectation of
-#' `sdlog` is the exponential of `eta_sdlog`.
-#'
-#' To alter the priors for this model, we suggest two possible workflows:
-#'
-#' 1. Use `brms::get_prior` to extract the default `brms` priors for your model.
-#' You may then alter these priors, and pass them to `epidist`. Note that the
-#' default `brms` priors are different to the priors we suggest.
-#'
-#' 2. Alternatively, after fitting the model, the all priors used in the model
-#' may be extracted using `brms::prior_summary()`.
-#'
-#' Examples of these two workflows will be provided as a part of a vignette at
-#' a future date.
+#' We provide suggested weakly informative priors.
 #'
 #' @inheritParams epidist_prior
 #' @method epidist_prior epidist_latent_individual
 #' @family latent_individual
 #' @importFrom cli cli_inform
 #' @export
-epidist_prior.epidist_latent_individual <- function(data, ...) {
+epidist_prior.epidist_latent_individual <- function(data, family, formula,
+                                                    ...) {
   epidist_validate(data)
-
-  prior1 <- brms::prior("normal(2, 0.5)", class = "Intercept")
-  prior2 <- brms::prior("normal(0, 0.5)", class = "Intercept", dpar = "sigma")
-
-  msg <- c(
-    "i" = "The following priors have been set:",
-    "*" = "normal(2, 0.5) on the intercept of distributional parameter mu",
-    "*" = "normal(0, 0.5) on the intercept of distributional parameter sigma",
-    "To alter priors, or set priors on other parameters, see ?epidist_prior."
-  )
-
-  cli::cli_inform(
-    message = msg, .frequency = "regularly", .frequency_id = "prior-message"
-  )
-
-  return(prior1 + prior2)
+  if (identical(family$dpars, c("mu", "sigma"))) {
+    prior_mu <- brms::prior("normal(2, 0.5)", class = "Intercept")
+    prior_sigma <- brms::prior(
+      "normal(0, 0.5)", class = "Intercept", dpar = "sigma"
+    )
+    msg <- c(
+      "i" = "The following priors have been set:",
+      "*" = "normal(2, 0.5) on the intercept of distributional parameter mu",
+      "*" = "normal(0, 0.5) on the intercept of distributional parameter sigma",
+      "To alter priors, or set priors on other parameters, see ?epidist_prior."
+    )
+    cli::cli_inform(
+      message = msg, .frequency = "regularly", .frequency_id = "prior-message"
+    )
+    priors <- prior_mu + prior_sigma
```

Comment:
Missing Gamma priors.

Reply:
There are no model priors in here. Add functionality for latent parameter priors at some point?